src/main/io/gimbal_control.c

```diff
@@ -0,0 +1,224 @@
+
+// This packet is sent to control the gimbal mode, sensivity and position
+#define GIMBAL_CMD_S0  0xA5
+#define GIMBAL_CMD_S1  0x5A
+typedef struct {
+    uint8_t  opcode[2];
+    uint64_t mode:3;     // Mode [0~7] Only 0 1 2 modes are supported
+    int64_t  sens:5;     // Sensitivity [-16~15]
+    uint64_t padding:4;
+    int64_t  roll:12;   // Roll angle [-2048~2047] - ±180deg
+    int64_t  pitch:12;  // Pich angle [-2048~2047] - ±180deg
+    int64_t  yaw:12;    // Yaw angle [-2048~2047]  - ±180deg
+    uint64_t crch:8;
+    uint64_t crcl:8;
+}  __attribute__ ((__packed__)) gimbalCmd_t;
+
+/* This packet is sent by the user to the head chase to realize the head
+ * chase calibration and reset function
+ */
+#define GIMBAL_CMD_L0  0xC5
+#define GIMBAL_CMD_L1  0x6E
+typedef struct {
+    uint8_t opcode[2];
+    uint8_t cmd;        // Command
+    uint8_t pwm:5;      // Pulse width data [0~31] => [0%~100%]
+    uint8_t iom:3;      // GPIO mode [0~7]
+    uint8_t mode:3;     // Mode [0~7] Only 0 1 2 modes are supported
+    int8_t  sens:5;     // Sensitivity [-16~15]
+    struct
+    {
+        uint32_t chan:3;  // Channel [0~7] [CH56,CH57,CH58,CH67,CH68,CH78,CH78,CH78]
+        uint32_t revs:2;  // Reverse [0~3] [Normal, Horizontal Reverse, Vertical Reverse, All Reverse]
+        uint32_t rngx:2;  // Range [0~3] [90 degrees, 120 degrees, 180 degrees, 360 degrees]
+        uint32_t rngy:2;  // Range [0~3] [60 degrees, 90 degrees, 120 degrees, 180 degrees]
+        int32_t  zerx:6;  // Zero [-32~31] [Resolution:5us]
+        int32_t  zery:6;  // Zero [-32~31] [Resolution:5us]
+        uint32_t padding:11;
+    } ppm;
+    uint8_t padding2[5];
+    uint64_t crch:8;
+    uint64_t crcl:8;
+}  __attribute__ ((__packed__)) gimbalCal_t;
+
+// Status reponse packet
+#define GIMBAL_STAT0  0x3A
+#define GIMBAL_STAT1  0x91
+typedef struct {
+    uint8_t opcode[2];
+    uint8_t cmd;        // Command response status
+    uint8_t ctyp;       // Calibration type [0:Idle 1:Acceleration calibration 2:Gyroscope calibration 3:Magnetometer calibration]
+    uint8_t cprg;       // Calibration progress [0%~100%]
+    uint8_t cerr;       // Calibration error [0%~100%]
+    uint8_t padding[8];
+    uint64_t crch:8;
+    uint64_t crcl:8;
+}  __attribute__ ((__packed__)) gimbalCalStatus_t;
+
+#define GIMBAL_ROLL_MIN -500
+#define GIMBAL_ROLL_MAX 500
+#define GIMBAL_PITCH_MIN -1150
+#define GIMBAL_PITCH_MAX 1750
+#define GIMBAL_YAW_MIN -2047
+#define GIMBAL_YAW_MAX 2047
+
+static gimbalCmd_t gimbalCmd = {0};
+static serialPort_t *gimbalSerialPort = NULL;
+
+static uint16_t gimbalCrc(uint8_t *buf, uint32_t size)
+{
+    return crc16_ccitt_update(0x0000, buf, size);
+}
+
+// Set the gimbal position on each axis in a ±500 range
+bool gimbalSet(int16_t roll, int16_t pitch, int16_t yaw)
+{
+    DEBUG_SET(DEBUG_GIMBAL, 0, roll);
+    DEBUG_SET(DEBUG_GIMBAL, 1, pitch);
+    DEBUG_SET(DEBUG_GIMBAL, 2, yaw);
+
+    if (!gimbalSerialPort) {
+        return false;
+    }
```

Comment:
```suggestion
```

Reply:
Should some code call `gimbalSet()` then it might as well return immediately if not valid gimbal port is opened.